### PR TITLE
Fix #827: Expose OTP resent period

### LINKED ISSUE #827: Expose OTP resent period
OTP resent period is configurable and may be changed. Mobile application needs to configure it as well. But these two values may be out of sync. Expose the value at REST API, where the mobile application may take it.

See `enrollment-server-onboarding.onboarding-process.otp.resend-period:30s`

### PR DESCRIPTION
**Pull request by jandusil**
- Update ConfigurationDataDto to expose otpResendPeriodSeconds
- Expose the Dto to responses of api/onboarding/start and api/onboarding/status endpoints
- Fill the config in services

### CODE REVIEW

File: enrollment-server-onboarding-api-model/src/main/java/com/wultra/app/enrollmentserver/api/model/onboarding/response/data/ConfigurationDataDto.java [hunk 1]
```diff
@@ -31,4 +31,9 @@ public class ConfigurationDataDto {
      */
     private String otpResendPeriod;
```
Comment (banterCZ) [inline enrollment-server-onboarding-api-model/src/main/java/com/wultra/app/enrollmentserver/api/model/onboarding/response/data/ConfigurationDataDto.java:32]: We may mark this property as deprecated, create an issue for removal and link it from here something like `// TODO (dusil, 2023-08-10, #xxx) some explanation`

File: enrollment-server-onboarding/src/main/java/com/wultra/app/onboardingserver/impl/service/OnboardingServiceImpl.java [hunk 2]
```diff
@@ -121,6 +127,9 @@ public OnboardingServiceImpl(
         this.otpService = otpService;
         this.activationService = activationService;
         this.onboardingProvider = onboardingProvider;
+        this.integrationConfigDto = new ConfigurationDataDto();
```
Comment (banterCZ) [inline enrollment-server-onboarding/src/main/java/com/wultra/app/onboardingserver/impl/service/OnboardingServiceImpl.java:130]: You may introduce a builder if you want so.

File: enrollment-server-onboarding-api-model/src/main/java/com/wultra/app/enrollmentserver/api/model/onboarding/response/OnboardingStartResponse.java [hunk 3]
```diff
@@ -30,5 +31,6 @@ public class OnboardingStartResponse {
 
     private String processId;
     private OnboardingStatus onboardingStatus;
+    private ConfigurationDataDto configurationDataDto;
```
Comment (banterCZ) [inline enrollment-server-onboarding-api-model/src/main/java/com/wultra/app/enrollmentserver/api/model/onboarding/response/OnboardingStartResponse.java:34]: The property name is not consistent with `IdentityVerificationStatusResponse`, where it is only `config`.

File: enrollment-server-onboarding-api-model/src/main/java/com/wultra/app/enrollmentserver/api/model/onboarding/response/OnboardingStatusResponse.java [hunk 4]
```diff
@@ -30,5 +31,6 @@ public class OnboardingStatusResponse {
 
     private String processId;
     private OnboardingStatus onboardingStatus;
+    private ConfigurationDataDto configurationDataDto;
```
Comment (banterCZ) [inline enrollment-server-onboarding-api-model/src/main/java/com/wultra/app/enrollmentserver/api/model/onboarding/response/OnboardingStatusResponse.java:34]: dtto

File: enrollment-server-onboarding/src/main/java/com/wultra/app/onboardingserver/impl/service/OnboardingServiceImpl.java [hunk 5]
```diff
@@ -84,6 +85,11 @@ public class OnboardingServiceImpl extends CommonOnboardingService {
 
     private final ActivationService activationService;
 
+    /**
+     * Configuration data for client integration
+     */
+    private final ConfigurationDataDto integrationConfigDto;
```
Comment (banterCZ) [inline enrollment-server-onboarding/src/main/java/com/wultra/app/onboardingserver/impl/service/OnboardingServiceImpl.java:91]: Ok, IMHO it is quite unusual but why not. In theory, this makes it impossible to reload properties in runtime, which we do not use anyway. Just a note to realize.

**Review by banterCZ — COMMENTED** [review]
Looks good from the backend point of view, except consistency of property naming. Also, let's wait for an opinion from mobile developers.

**Review by kober32 — APPROVED** [review]
OK with me.

**Pull request merged**